src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/SessionParameterValueResolver.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\MapSessionParameter;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+final class SessionParameterValueResolver implements ValueResolverInterface
+{
+    public function resolve(Request $request, ArgumentMetadata $argument): array
+    {
+        if (!$request->hasSession()) {
+            return [];
+        }
+
+        if (!$argument->getAttributesOfType(MapSessionParameter::class, ArgumentMetadata::IS_INSTANCEOF)) {
+            return [];
+        }
+
+        if ((!$type = $argument->getType()) || !class_exists($type)) {
```

Comment:
we could now allow an interface here: providing a default value to the argument means we know how to build the initial value:
```suggestion
        if ((!$type = $argument->getType()) || (!class_exists($type) && !interface_exists($type, false))) {
```